src/discussions/comments/data/slices.js

```diff
@@ -20,26 +20,31 @@ const commentsSlice = createSlice({
     // TODO: save in localstorage so user can continue editing?
     commentDraft: null,
     postStatus: RequestStatus.SUCCESSFUL,
-    pagination: {
-    },
+    pagination: {},
+    responsesPagination: {},
   },
   reducers: {
     fetchCommentsRequest: (state) => {
       state.status = RequestStatus.IN_PROGRESS;
     },
     fetchCommentsSuccess: (state, { payload }) => {
+      const { threadId } = payload;
+      // force endorsed to be null, true or false
+      const endorsed = payload.endorsed === undefined ? null : payload.endorsed;
       state.status = RequestStatus.SUCCESSFUL;
-      state.commentsInThreads[payload.threadId] = [
-        ...(state.commentsInThreads[payload.threadId] || []),
-        ...(payload.commentsInThreads[payload.threadId] || []),
+      state.commentsInThreads[threadId] = state.commentsInThreads[threadId] ?? {};
+      state.pagination[threadId] = state.pagination[threadId] ?? {};
+      state.commentsInThreads[threadId][endorsed] = [
+        ...(state.commentsInThreads[threadId][endorsed] || []),
+        ...(payload.commentsInThreads[threadId] || []),
```

Comment:
when you assigned null to endorsed and then you assign values here it makes a weird redux structure like this
<img width="600" alt="Screenshot 2021-09-30 at 7 16 46 PM" src="https://user-images.githubusercontent.com/67791278/135472646-5adcfbce-e753-4400-8ef4-81d48984e0a6.png">


I dont understand the logic here? dont we have some other solution to differentiate between endorsed and unendorsed comments?

Reply:
@mehaknasir 
@xitij2000 can correct me if I'm wrong, but I think the idea here is such: questions have endorsed and unendorsed comments, and discussions don't have such differentiation. So when the thread (aka post) is a question, `endorsed` can have values `true` and `false`, and for threads that are discussions we need to set `endorsed` to a value which signifies the absence of differentiation, and since `null` signifies the absence of value, I think it's suitable to use it in this manner.
I'm sure there are alternative approaches, and I also think current solution is one of the cases where `null` is appropriately used for semantics.

Reply:
Btw, I need to mention, that these changes are from #22, I just rebased my branch onto it, because there are some changes that this PR depends on, so may be it's more appropriate to discuss it there.

Reply:
We can as easily create a mapping where we use:
"endorsed" instead of true
"unendorsed" instead of false
"discussion" instead of null

However, since the redux state is entirely internal and not seen by users I didn't see much reason for adding this complexity.

Reply:
Even if its related to internal implementation I would still like to have some meaningful structuring of redux data. for maintaining code readability and quality. I would suggest changing this.

Reply:
I'm afraid I disagree here. It's the code that should be readable here, not the data. We're dealing with raw ids like "906e91ed8a464499aef4561829", none of that is supposed to be readble. It is supposed to be structured in a way to allow easy access to data. That is why the data is being restructured from the original API response to something that makes queries more efficient. 

I don't understand how the code will become more readable, or how its quality will improve if instead of a simple operation like:

``commentsInThreads[threadId][endorsed] ``

we now have:

``commentsInThreads[threadId][ENDORSED_NAME_MAP[endorsed]]``

To me, the new code above is a lot **less** readable (whatever you call the variable instead of `ENDORSED_NAME_MAP`). Every time we need to fetch or save this data we'll have to map this value.

Reply:
@xitij2000 PR is looking good. I totally agree with @mehaknasir suggestion. Assigning a readable name instead of null will increase more readable structure. We also posted an open suggestion question related to this practice in fedx. We will approve this PR soon.

Reply:
(please excuse the intrusion)

Under the current suggestions, you would still use `commentsInThreads[threadId][endorsed]`, however, `endorsed` would generally be a string value, either `'endorsed'`, `'unendorsed'`, or `'discussion'`. As a result, the object keys will be human readable.

Reply:
> Under the current suggestions, you would still use commentsInThreads[threadId][endorsed], however, endorsed would generally be a string value, either 'endorsed', 'unendorsed', or 'discussion'.
@kenclary 
Here is the thing, endorsed is a value returned from the API, its value is either `true`, or `false`. Its value isn't "endorsed" or "unendorsed", so we'll need to make the mapping at some point. I just don't see the reason why. 

This is a data structure that is only ever accessed by the code, humans do not look at it unless they install a specific browser extension and use it to inspect the state when the code is running in a development environment. 

I really don't understand the need for human-readable data structures here. We don't inspect the value of each python variable in code to make sure that it's contents are human-readable. We see if the code is readable, and if the data is structured in a way that allows efficient access. 

I could understand the argument if this was supposed to be saved to a file and read by humans at any point. I mean, we can go further here and use the title of the post instead of its id, as that will be more human-readable. 

I don't want to endlessly block this work, so I'll update my PR, but I don't think human readability should ever be a concern when creating temporary data structures that will never be seen by a human outside of debugging. If this is to become a recommended frontend or backend practice then I would strongly urge against it.

Reply:
I've updated the code here: https://github.com/edx/frontend-app-discussions/pull/22/commits/07dc355e231c7397a84a81304004c0a5cbcde210 
@Cup0fCoffee Could you rebase on my latest changes once they are approved.